File: src/main/java/org/application/web/libs/observer/Subject.java
package org.application.web.libs.observer;

import java.util.List;
import java.util.ArrayList;

public class Subject {
    private List<Observer> handlers;

    public Subject() {
        handlers = new ArrayList<>();
    }

    public void subscribe(Observer observer) {
        handlers.add(observer);
    }

    public void notify(Object data) {
        // for (Observer handler : handlers) {
        //     handler.notify(data);
        // }
    }
}

File: Source/Utils/Observer/Subject.cs
namespace Application.Source.Utils.Observer
{
    public abstract class Subject
    {
        private readonly List<IObserver> handler;

        protected Subject()
        {
            handler = [];
        }

        public void Subscribe(IObserver observer)
        {
            handler.Add(observer);
        }

        public void Unsubscribe(IObserver observer)
        {
            handler.Remove(observer);
        }

        public void Notify()
        {
            foreach (var observer in handler)
            {
                observer.Update();
            }
        }
    }
}